core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java

```diff
@@ -37,6 +42,37 @@
         this.relationName = relationName;
     }
 
+    @Override
+    public ShapeRelation readFrom(StreamInput in) throws IOException {
```

Comment:
folks are generally concnered that the ordinal might cahnge if you change the order of the enum. I usually write explicit tests for it but with switch / case it's declerative and that should protect from it?